src/ParticleDA.jl

```diff
@@ -336,10 +336,10 @@ function get_covariance_observation_state_given_previous_state(model_data)
     cov = Matrix{get_state_eltype(model_data)}(
         undef, length(state_indices), observation_dimension
     )
-    for i in state_indices
+    for i in length(state_indices)
```

Comment:
This doesn't look correct as the loop is now over an integer (the length of `state_indices`) rather than a collection. I think something like
```Julia
for (i, state_index) in enumerate(state_indices)
```
is probably more what we want here and use `i` to index in to the `cov` array and pass `state_index` to the `get_covariance_state_observation_given_previous_state` function?

Reply:
Yes the enumerate will work, will update the PR now. We were getting a bounds error when solely observing a variable that isn't the first in the flattened array.